### ISSUE TITLE
Spain and UK recovered data need compliment -> improve complements

### ISSUE THREAD

**Issue by Inglezos** [lisphilar/covid19-sir#427]
## Summary
Scenario analysis for Spain and UK after #425, is problematic because both countries have irregular recovered data which are not updated after some point (value is irrelevant). A third condition to indicate full complement is needed, which will capture the fact that recovered data are not currently updated (for example same max recovered value for over two weeks). Since they are not updated we can invalidate them and generate fully complemented, keeping ideally the first valid ones.
Also a fourth condition is needed in order to ensure that recovered are not too low-valued (at least 1% of confirmed minus fatal), for example to cover UK situation.

To sum up, we can distinguish four cases for full complement need:
1. No recovered records (such as China, Netherlands)
2. Too low-valued recovered data compared to confirmed (such as UK)
3. Too high-valued recovered data compared to confirmed (such as Germany)
4. Not updated currently for over two weeks, full compliment only these values and keep the previous valid values (such as Spain, Greece, etc). This case could be covered if we run partial complement right before full complement, but this will lead to low values compared to the confirmed, but not too low in order to be captured by 2. Thus, they should be handled in a new function for partial complement ending.

## Code:
```
import covsirphy as cs
spa_scenario = cs.Scenario(jhu_data, population_data, "Spain")
spa_scenario.records().tail()
uk_scenario = cs.Scenario(jhu_data, population_data, "UK")
uk_scenario.records().tail()
```

**Comment by Inglezos** [comment]
Created pull request #428.

**Comment by lisphilar** [comment]
Thank you for creating the pull request. I added some comments.

**Comment by Inglezos** [comment]
I have reworked the findings, please review again and merge.

**Comment by lisphilar** [comment]
Thank you for updated codes!
I merged your pull request and this issue will be fixed in stable version 2.13.2
